src/fvm_multicell.hpp

```diff
@@ -1075,13 +1087,25 @@ void fvm_multicell<Backend>::reset() {
     set_time_global(0);
     set_time_to_global(0);
 
+    // Update ion species:
+    //   - clear currents
+    //   - reset concentrations to defaults
+    //   - recalculate reversal potentials
+    for (auto& i: ions_) {
+        i.second.reset();
+    }
+
     for (auto& m : mechanisms_) {
-        // TODO : the parameters have to be set before the nrn_init
-        // for now use a dummy value of dt.
         m->set_params();
         m->nrn_init();
     }
 
+    // Updeate reversal potential to account for changes to concentrations made
+    // by calls to nrn_init() in mechansisms.
+    for (auto& i: ions_) {
+        i.second.update_reversal_potential(6.3+273.15); // TODO: use temperature specfied in model
```

Comment:
I'd pop this TODO into a ticket, or extend the `arb::cell_global_properties` (in `cell.hpp`) to include the temperature field and assign this from that.

Reply:
Already done: #386